### DISCUSSION TITLE
Is it possible to set a positive gain?

### DISCUSSION THREAD

**Discussion by DrewNeilson** [mrene/minidsp-rs#278] [Q&A]
Is it possible to set a positive value for either the master gain or for any of the individual input or output gains? So far, in the documentation I see information that most gains are negative, but I'd like to be able to set a positive master gain or, if not that, for each of the individual input channels or for each of the individual output channels. My MiniDSP is a 2x4HD.

**Comment by mrene** [comment]
Yes as long as you follow the same supported ranges as the plugin, positive gains will work.

**Reply by DrewNeilson** [reply]
Can you show the exact syntax to use to set a positive master gain, a positive input channel gain, and a positive output channel gain? Every time that I try to set a positive gain, the value gets entered into the MiniDSP as negative.

**Reply by mrene** [reply]
I just checked the 2x4HD plugin, the master gain is limited between -127.5dB and 0dB, so positive gains aren't possible there, but you can set a positive input and output gain up to 12dB.

`minidsp input 0 gain 12` would do as such for the first input channel - note that you will not see this updated in their app because the devices are write-only.

**Reply by DrewNeilson** [reply]
> I just checked the 2x4HD plugin, the master gain is limited between -127.5dB and 0dB, so positive gains aren't possible there, but you can set a positive input and output gain up to 12dB.
> 
> `minidsp input 0 gain 12` would do as such for the first input channel - note that you will not see this updated in their app because the devices are write-only.

In the Windows 10 Command Prompt, I used the command that you wrote, and did not get the desired outcome.
I started in the following directory, which is the folder in which I keep minidsp.exe.
```
C:\self-contained applications\minidsp-rs\pre-built>
```
I typed
```
minidsp
```
Here is the response.
```
MasterStatus { preset: 1, source: Analog, volume: Gain(-0.0), mute: false, dirac: false }
Input levels: -99.6, -120.0
Output levels: -99.9, -100.1, -100.1, -120.0
```
I was returned to
```
C:\self-contained applications\minidsp-rs\pre-built>
```
I typed
```
minidsp input 0 gain 12
```
After a few seconds I was returned to
```
C:\self-contained applications\minidsp-rs\pre-built>
```
I typed
```
minidsp
```
Here is the response:
```
MasterStatus { preset: 1, source: Analog, volume: Gain(-0.0), mute: false, dirac: false }
Input levels: -88.1, -120.0
Output levels: -88.2, -88.3, -88.3, -120.0
```

**Reply by mrene** [reply]
only the master gain is reported when running `minidsp` without arguments (and this one must be between -127.5dB and 0dB). the individual input and output channel gains aren't reported because they can't be read from the device, only set.

**Reply by DrewNeilson** [reply]
> only the master gain is reported when running `minidsp` without arguments (and this one must be between -127.5dB and 0dB). the individual input and output channel gains aren't reported because they can't be read from the device, only set.

At
```
C:\self-contained applications\minidsp-rs\pre-built>
```
I typed
```
minidsp gain 12
```
I was returned to
```
C:\self-contained applications\minidsp-rs\pre-built>
```
I typed
```
minidsp gain 12
```
I was returned to
```
C:\self-contained applications\minidsp-rs\pre-built>
```
I typed
```
minidsp
```
and here is the response:
```
MasterStatus { preset: 1, source: Analog, volume: Gain(-12.0), mute: false, dirac: false }
Input levels: -88.4, -120.0
Output levels: -100.5, -100.6, -100.6, -120.0
```
So, I get a master gain of -12.0 dB. How do I set a positive master gain?

**Reply by mrene** [reply]
You can't, even their apps don't allow you to do that, it's limited between -127.5dB and 0dB for the master gain. Only individual input/output channels can be set to +12dB.
